rclcpp_lifecycle/src/lifecycle_node_interface_impl.cpp

```diff
@@ -182,6 +182,8 @@ LifecycleNode::LifecycleNodeInterfaceImpl::init(bool enable_communication_interf
         nullptr);
     }
   }
+
+  current_state_ = State(state_machine_.current_state);
```

Comment:
> Does this have to be after `enable services` above? i think this can be right after `rcl_lifecycle_state_machine_init`?

Yeah, that would be slightly more correct.  It would only matter in the case that `rcl_lifecycle_state_machine_init` succeeded, but one of the `add_service` calls above failed.  I can move it (though it is minor enough that I may just add it to #1756 rather than making a new PR for it).

> and probably should be updated on `rcl_lifecycle_state_machine_fini` as well?

I guess we could, but I don't think it really matters; `fini` is only called in the destructor, so we are about to destroy the class anyway.

Reply:
agree, resolving this thread